packages/puppeteer-crawler/src/internals/utils/puppeteer_utils.ts

```diff
@@ -191,6 +191,26 @@ export async function injectJQuery(page: Page, options?: { surviveNavigations?:
 export async function parseWithCheerio(page: Page, ignoreShadowRoots = false): Promise<CheerioRoot> {
     ow(page, ow.object.validate(validators.browserPage));
 
+    if (page.frames().length > 1) {
```

Comment:
The thing is, `@crawlee/playwright` and `@crawlee/puppeteer` are separate packages, so we would have to create a new package for this shared code (any other crawlee package doesn't / cannot depend on `playwright` or `puppeteer`(?)). 

I see that these two are verbatim copies, but that's only because here we're using the subsets of PW / PP interfaces that are equal... other utils methods are different for PW / PP. I like to think of these as "platform" specific ports of the same features.

Reply:
Couldn't it be put in `@crawlee/browser-crawler` somehow?

Reply:
Because of what I mentioned above, it would be very awkward - see here:

https://github.com/apify/crawlee/blob/9918747961324f86e5789f3d1acecde371493fcd/packages/browser-crawler/src/internals/browser-crawler.ts#L818-L823

Or here:

https://github.com/apify/crawlee/blob/9918747961324f86e5789f3d1acecde371493fcd/packages/browser-pool/src/abstract-classes/browser-plugin.ts#L42-L45

Dependency injection... or something, I guess.

With this as an alternative, I'm more than happy to have "duplicate" separate implementations for both libraries.

Reply:
Hm, I guess you'd have to write quite a lot of boilerplate types. I guess I'm equally unhappy with both approaches.

Reply:
`@crawlee/browser` package has optional peer dependencies on both playwright and puppeteer, so you can surely have a code that works with both of them inside it. But to do that without hacks like ts-ignore comments and dynamic imports, you would need to introduce separate exports for each library that wouldn't be exported from the root index file. Probably not worth it now.